src/components/PostPageDetail/index.js

```diff
@@ -112,12 +112,11 @@ const PostPageDetailIdComponent = (props) => {
   };
   const initial = () => {
     const reactionCount = item?.reaction_counts;
-    if (JSON.stringify(reactionCount) !== '{}') {
+    if (JSON.stringify(reactionCount) !== '{}' && reactionCount) {
       let count = 0;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The check for `reactionCount` being truthy should be done before stringifying it. This will prevent unnecessary operations when `reactionCount` is null or undefined.
```diff
- if (JSON.stringify(reactionCount) !== '{}' && reactionCount) {
+ if (reactionCount && JSON.stringify(reactionCount) !== '{}') {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->